File: apps/api/graphql_types/onboarding.py
from enum import Enum
from typing import List, Optional

import strawberry

from .auth import Organization


@strawberry.enum
class RowStatus(Enum):
    SUCCESS = "SUCCESS"
    ERROR = "ERROR"
    SKIPPED = "SKIPPED"


@strawberry.input
class BulkSetupRow:
    row_id: str
    property_name: str
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    phone: Optional[str] = None


@strawberry.input
class BulkSetupInput:
    organization_name: str
    rows: List[BulkSetupRow]


@strawberry.type
class BulkSetupRowResult:
    row_id: str
    status: RowStatus
    error: Optional[str] = None
    property_id: Optional[str] = None
    user_id: Optional[str] = None


@strawberry.type
class BulkSetupResult:
    organization: Organization
    total_properties: int
    total_residents: int
    rows: List[BulkSetupRowResult]

File: apps/web/lib/queries/onboarding.ts
export const BULK_SETUP_ORGANIZATION = `
  mutation BulkSetupOrganization($input: BulkSetupInput!) {
    bulkSetupOrganization(input: $input) {
      organization {
        id
        name
        slug
      }
      totalProperties
      totalResidents
      rows {
        rowId
        status
        error
        propertyId
        userId
      }
    }
  }
`;

export const COMPLETE_PROFILE = `
  mutation CompleteProfile($input: CompleteProfileInput!) {
    completeProfile(input: $input) {
      id
      firstName
      lastName
      email
      phone
    }
  }
`;

export type BulkSetupRow = {
  rowId: string;
  propertyName: string;
  firstName?: string;
  lastName?: string;
  phone?: string;
};

export type BulkSetupInput = {
  organizationName: string;
  rows: BulkSetupRow[];
};

export type RowStatus = "SUCCESS" | "ERROR" | "SKIPPED";

export type BulkSetupRowResult = {
  rowId: string;
  status: RowStatus;
  error?: string;
  propertyId?: string;
  userId?: string;
};

export type BulkSetupResult = {
  bulkSetupOrganization: {
    organization: {
      id: string;
      name: string;
      slug: string;
    };
    totalProperties: number;
    totalResidents: number;
    rows: BulkSetupRowResult[];
  };
};
